articles/server-platforms/rails/02-custom-login.md

```diff
@@ -0,0 +1,225 @@
+
+Next, add the definitions for the `login` and `signup` actions under `private`:
+
+```ruby
+def login
+  token = client.login(
+    params[:user],
+    params[:password],
+    authParams: {
+     scope: 'openid name email'
+    },
+    connection:'Username-Password-Authentication'
+  )
+
+  session[:token_id] = token
+end
+
+def signup
+  token = client.signup(
+    params[:user],
+    params[:password]
+  )
+end
+```
+
+### 4. Add an SDK client instance in the Application Controller
+In order to call the API methods using the Auth0 Ruby SDK, you need to initialize a client instance. Add the following code in your Application Controller (`controllers/application_controller.rb`)
+
+```ruby
+def client
+  creds = { client_id: ENV['AUTH0_CLIENT_ID'],
+    client_secret: ENV['AUTH0_CLIENT_SECRET'],
+    api_version: 1,
+    domain: ENV['AUTH0_DOMAIN'] }
+
+  @client = Auth0Client.new(creds)
+end
+```
+
+### 5. Checking if the user is authenticated
+In [Step 01](/quickstart/webapp/rails/01-login) you used a parent controller with a `before_action` to prevent users from executing actions in controllers that inherited from it. You will use the same strategy now, but the code will change slightly.
+
+Create a `SecuredController` class with the code below:
+
+```ruby
+class SecuredController < ApplicationController
+  before_action :logged_in?
+
+  private
+
+  def logged_in?
+    unless session[:token_id].present?
+      redirect_to '/'
+    end
+  end
+end
+```
+
+Then, update the `DashboardController` so that it inherits from `SecuredController`
+
+```ruby
+class DashboardController < SecuredController
+```
+
+This way, if a user attempts to access the routes governed by the `DashboardController` actions, it will be redirected to `/`.
+
+### 6. Login with Google
+
+For your users to login using a Social Connection without using neither Lock nor OmniAuth, follow these steps:
+
+> **Note**: Social connections support only browser based (passive) authentication. The providers don't allow entering the user/password in applications that don't belong to them, so the user gets redirected to the provider (e.g. Google sign in page). 3 steps take place in this process: Initiation, Authentication, and Getting the Access Token. You'll trigger the Initiation, Auth0 will take care of the Authentication, and then you'll obtain the access and id token for the user (*to learn more about this process go to: https://auth0.com/docs/api/authentication#!%23get--authorize_social and https://auth0.com/docs/protocols#oauth-server-side*)
+
+For this particular example, you'll configure Google as your social provider. The API methods called by the API also support Facebook, Twitter and Weibo.
+
+**Prerequisite**: configure your Google Social Connection by following [this tutorial](https://auth0.com/docs/connections/social/google).
```

Comment:
URLs should be relative:

`https://auth0.com/docs/connections/social/google` should be `/connections/social/google`